modules/clients/src/test/java/org/apache/ignite/internal/jdbc2/JdbcQueryQuotaTest.java

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.jdbc2;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.cache.query.SqlFieldsQuery;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.query.SqlFieldsQueryEx;
+import org.apache.ignite.internal.processors.query.oom.DiskSpillingAbstractTest;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Test;
+
+/**
+ * Jdbc test for query quota configuration.
+ */
+@SuppressWarnings("ThrowableNotThrown")
+public class JdbcQueryQuotaTest extends DiskSpillingAbstractTest {
+    /** */
+    private static final String QUERY_512_TO_1024 = "SELECT DISTINCT id FROM person WHERE id < 8";
+
+    /** */
+    private static final String QUERY_1024_TO_2048 = "SELECT DISTINCT id FROM person WHERE id < 16";
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setSqlOffloadingEnabled(IgniteConfiguration.DFLT_SQL_QUERY_OFFLOADING_ENABLED);
+        cfg.setSqlQueryMemoryQuota("1024");
+        cfg.setSqlGlobalMemoryQuota(IgniteConfiguration.DFLT_SQL_QUERY_GLOBAL_MEMORY_QUOTA);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected boolean startClient() {
+        return false;
+    }
+
+    /**
+     * Test for clients
+     * Start server node with default configuration and QueryMemoryQuota set to 1024
+     * Start client of type Client type with default configuration and QueryMemoryQuota set to 2048
+     * Execute SQL that would take > 1024 and < 2048 bytes for result
+     * Ensure that SqlException query quota exceeded is thrown
+     * Start client of type Client type with default configuration and QueryMemoryQuota set to 512
+     * Execute SQL that would take > 512 and < 1024 bytes for result
+     * Ensure that SqlException query quota exceeded is thrown
+     */
+    @Test
+    public void testClientQueryQuota() throws Exception {
+        try (Connection conn512 = createConnection("jdbc-config-query-mem-limit-512.xml");
+             Connection conn2048 = createConnection("jdbc-config-query-mem-limit-2048.xml")) {
+
+            GridTestUtils.assertThrows(log, () -> {
+                Statement stmt = conn2048.createStatement();
+
+                stmt.execute(QUERY_1024_TO_2048);
+            }, IgniteException.class, "SQL query run out of memory: Query quota exceeded.");
```

Comment:
could we add additional check with no fail for QUERY_512_TO_1024?

Reply:
Done